contracts/FlowIDTableStaking.cdc

```diff
@@ -1012,15 +1014,21 @@ pub contract FlowIDTableStaking {
                 if currentRoleCount > 0 {
                     currentRoleNodeCounts[nodeRecord.role] = currentRoleCount - 1
                 }
-                FlowIDTableStaking.account.load<{UInt8: UInt16}>(from: /storage/flowStakingRoleNodeCounts)
-                FlowIDTableStaking.account.save(currentRoleNodeCounts, to: /storage/flowStakingRoleNodeCounts)
+                FlowIDTableStaking.account.storage.load<{UInt8: UInt16}>(from: /storage/flowStakingRoleNodeCounts)
+                FlowIDTableStaking.account.storage.save(currentRoleNodeCounts, to: /storage/flowStakingRoleNodeCounts)
             }
 
-            var movesPendingList = FlowIDTableStaking.account.borrow<&{String: {UInt32: Bool}}>(from: /storage/idTableMovesPendingList)
+            var movesPendingList = FlowIDTableStaking.account.storage.borrow<auth(Mutate) &{String: {UInt32: Bool}}>(from: /storage/idTableMovesPendingList)
                 ?? panic("No moves pending list in account storage")
 
             // Iterate through all delegators and unstake their tokens
-            for delegator in nodeRecord.delegators.keys {
+            // since their node has unstaked
+            let keys = nodeRecord.delegators.keys
```

Comment:
Seems like this iteration and the logic within it is repeated a couple times. Could this be made into a helper or maybe an iterator like `forEachDelegator`? Not pressing at all, maybe something to think about as a change to include later.

Reply:
Good idea! I created an issue for later: https://github.com/onflow/flow-core-contracts/issues/431